lib/bcu.rb

```diff
@@ -4,11 +4,58 @@
 require "cask/all"
 require "extend/formatter"
 require "extend/cask"
+require "fileutils"
+require "set"
+
+PINS_FILE = File.expand_path(File.dirname(__FILE__) + "/../pins")
 
 module Bcu
   def self.process(args)
     parse!(args)
 
+    FileUtils.touch(PINS_FILE)
+
+    pins = Set[]
+    File.open(PINS_FILE, "r") do |f|
+      f.each_line do |app|
+        pins.add(app.rstrip)
+      end
+    end
+
+    if options.list_pins
+      pins.each do |app|
+        puts app
+      end
+      return
+    end
+
+    if options.pin
+      if pins.include?(options.pin)
+        puts "Already pinned: #{options.pin}"
+      else
+        File.open(PINS_FILE, "a") do |f|
+          f.puts(options.pin)
+        end
+        puts "Pinned: #{options.pin}"
+      end
+      return
+    end
+
+    if options.unpin
+      if pins.include?(options.unpin)
+        pins.delete(options.unpin)
```

Comment:
Would be nice to extract that to a separate function that takes care of that.
```suggestion
        remove_pin options.unpin
```

The function is already quite big and if we could extract something out to increase readability, would be great. What do you think 🙂